### PR TITLE
feat: claude adaptive thinking

### PR DESCRIPTION
**Pull request by heiwen**
## Summary by CodeRabbit

* **New Features**
  * Added a "max" reasoning-effort option across models; treated as an alias for high/xhigh and included in effort-to-budget mapping.

* **Provider Support**
  * Bedrock now translates provider thinking/effort settings into a standardized Anthropic reasoning configuration and is registered for relevant models.
  * Bedrock set as an available provider in e2e setup with updated model catalog.

* **Tests**
  * Significantly expanded tests covering Claude effort mappings, budget clamping, max_tokens, and many edge cases.

### CODE REVIEW

File: src/models/anthropic/middleware.ts [hunk 1]
```diff
@@ -28,22 +77,44 @@ export const claudeReasoningMiddleware: LanguageModelMiddleware = {
...
+        // FUTURE: warn that reasoning.max_tokens was computed
+        target["thinking"] = {
+          type: "enabled",
+          budgetTokens: calculateReasoningBudgetFromEffort(
+            reasoning.effort,
+            params.maxOutputTokens ?? getMaxOutputTokens(modelId),
+            1024,
+          ),
+        };
+      }
```
[the remark this answers is not in the document]
Comment (heiwen) [reply src/models/anthropic/middleware.ts]: Good catch. In AI SDK it should be in "effort" actually.

File: src/models/anthropic/middleware.ts [hunk 2]
```diff
@@ -1,21 +1,70 @@
...
+}
+
 function getMaxOutputTokens(modelId: string): number {
   if (!modelId.includes("opus-4")) return CLAUDE_MAX_OUTPUT_TOKENS;
-  if (modelId.includes("opus-4.5") || modelId.includes("opus-4-5")) {
+  if (modelId.includes("opus-4.5")) {
     return CLAUDE_MAX_OUTPUT_TOKENS;
   }
   return CLAUDE_OPUS_4_MAX_OUTPUT_TOKENS;
 }
```
[the remark this answers is not in the document]
Comment (heiwen) [reply src/models/anthropic/middleware.ts]: Good catch, fixed

**Review by buibaoanh — APPROVED** [review]

File: src/providers/bedrock/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      const thinkingOptions = thinking as Record<string, unknown>;
+      if (thinkingOptions["type"] !== undefined) {
+        target["type"] = thinkingOptions["type"];
+      }
+      if (thinkingOptions["budgetTokens"] !== undefined) {
+        target["budgetTokens"] = thinkingOptions["budgetTokens"];
+      }
+    }
+
+    if (effort !== undefined) target["maxReasoningEffort"] = effort;
```
[the remark this answers is not in the document]
Comment (heiwen) [reply src/providers/bedrock/middleware.ts]: Good question, but unrelated to this PR. Let's tackle independently.

File: src/models/anthropic/middleware.ts [hunk 4]
```diff
@@ -1,21 +1,66 @@
...
+  ChatCompletionsReasoningConfig,
+  ChatCompletionsReasoningEffort,
+} from "../../endpoints/chat-completions/schema";
 
 import { modelMiddlewareMatcher } from "../../middleware/matcher";
 import { calculateReasoningBudgetFromEffort } from "../../middleware/utils";
 
-const CLAUDE_MAX_OUTPUT_TOKENS = 64000;
-const CLAUDE_OPUS_4_MAX_OUTPUT_TOKENS = 32000;
+const isClaude = (family: "opus" | "sonnet", version: string) => {
```
[the remark this answers is not in the document]
Comment (heiwen) [reply src/models/anthropic/middleware.ts]: Good catch, done
